File: src/main/java/work/xeltica/craft/core/commands/CommandHub.java
package work.xeltica.craft.core.commands;

import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import work.xeltica.craft.core.models.HubType;
import work.xeltica.craft.core.stores.HubStore;

import java.util.List;
/**
 * ロビーへ移動するコマンド
 * @author Xeltica
 */
public class CommandHub extends CommandPlayerOnlyBase {

    @Override
    public boolean execute(Player player, Command command, String label, String[] args) {
        final var store = HubStore.getInstance();

        // 現在いるワールドを踏まえてどのワールドに移動するかどうかを分岐する
        store.teleport(player, switch (player.getWorld().getName()) {
            default -> HubType.Main;
            case "hub" -> HubType.Classic;
            case "world" -> HubType.Classic;
            case "world_nether" -> HubType.Classic;
            case "world_the_end" -> HubType.Classic;
            case "nightmare" -> HubType.Classic;
            case "sandbox" -> HubType.Classic;
            case "wildarea" -> HubType.Classic;
        });

        return true;
    }

    @Nullable
    @Override
    public List<String> onTabComplete(@NotNull CommandSender commandSender, @NotNull Command command, String label, String[] args) {
        return COMPLETE_LIST_EMPTY;
    }
}

File: src/main/java/work/xeltica/craft/core/commands/CommandHub.kt
package work.xeltica.craft.core.commands

import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import work.xeltica.craft.core.api.commands.CommandPlayerOnlyBase
import work.xeltica.craft.core.modules.HubModule
import work.xeltica.craft.core.models.HubType

/**
 * ロビーへ移動するコマンド
 * @author Xeltica
 */
class CommandHub : CommandPlayerOnlyBase() {
    override fun execute(player: Player, command: Command, label: String, args: Array<String>): Boolean {
        HubModule.teleport(player, HubType.Main)
        return true
    }

    override fun onTabComplete(commandSender: CommandSender, command: Command, label: String, args: Array<String>): List<String>? {
        return COMPLETE_LIST_EMPTY
    }
}